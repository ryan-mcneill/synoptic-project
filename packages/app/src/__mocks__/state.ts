export const mockState = {
  albums: {
    loading: false,
    data: {
      "5f1423c54ee369215e05625e": {
        name: "Sample Album A",
        songs: [
          {
            _id: "5f14218d4ee369215e05624f",
            name: "Hey",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625e",
                name: "Sample Album A"
              }
            ]
          },
          {
            _id: "5f1422054ee369215e056256",
            name: "Cute",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625e",
                name: "Sample Album A"
              }
            ]
          },
          {
            _id: "5f1422054ee369215e056257",
            name: "Going Higher",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625e",
                name: "Sample Album A"
              }
            ]
          },
          {
            _id: "5f1ad1ef17e08f691bb74048",
            name: "Creative Minds",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              },
              {
                _id: "5f14228b4ee369215e05625b",
                name: "BenSound"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625e",
                name: "Sample Album A"
              }
            ]
          }
        ]
      },
      "5f1423c54ee369215e05625f": {
        name: "Sample Album B",
        songs: [
          {
            _id: "5f1422054ee369215e056255",
            name: "A New Beginning",
            artists: [
              {
                _id: "5f14228b4ee369215e05625b",
                name: "BenSound"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625f",
                name: "Sample Album B"
              }
            ]
          },
          {
            _id: "5f1422054ee369215e056258",
            name: "Ukulele",
            artists: [
              {
                _id: "5f14228b4ee369215e05625b",
                name: "BenSound"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625f",
                name: "Sample Album B"
              }
            ]
          },
          {
            _id: "5f1ad1ef17e08f691bb74048",
            name: "Creative Minds",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              },
              {
                _id: "5f14228b4ee369215e05625b",
                name: "BenSound"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625f",
                name: "Sample Album B"
              }
            ]
          }
        ]
      }
    },
    error: null
  },
  artists: {
    loading: false,
    data: {
      "5f14228b4ee369215e05625a": {
        name: "Benjamin Tissot",
        songs: [
          {
            _id: "5f14218d4ee369215e05624f",
            name: "Hey",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625e",
                name: "Sample Album A"
              }
            ]
          },
          {
            _id: "5f1422054ee369215e056256",
            name: "Cute",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625e",
                name: "Sample Album A"
              }
            ]
          },
          {
            _id: "5f1422054ee369215e056257",
            name: "Going Higher",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625e",
                name: "Sample Album A"
              }
            ]
          },
          {
            _id: "5f1ad1ef17e08f691bb74048",
            name: "Creative Minds",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              },
              {
                _id: "5f14228b4ee369215e05625b",
                name: "BenSound"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625e",
                name: "Sample Album A"
              }
            ]
          }
        ]
      },
      "5f14228b4ee369215e05625b": {
        name: "BenSound",
        songs: [
          {
            _id: "5f1422054ee369215e056255",
            name: "A New Beginning",
            artists: [
              {
                _id: "5f14228b4ee369215e05625b",
                name: "BenSound"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625f",
                name: "Sample Album B"
              }
            ]
          },
          {
            _id: "5f1422054ee369215e056258",
            name: "Ukulele",
            artists: [
              {
                _id: "5f14228b4ee369215e05625b",
                name: "BenSound"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625f",
                name: "Sample Album B"
              }
            ]
          },
          {
            _id: "5f1ad1ef17e08f691bb74048",
            name: "Creative Minds",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              },
              {
                _id: "5f14228b4ee369215e05625b",
                name: "BenSound"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625e",
                name: "Sample Album A"
              }
            ]
          }
        ]
      }
    },
    error: null
  },
  currentSong: {
    isPlaying: false,
    background: {
      isDark: false,
      colour: "#e0c1e2"
    }
  },
  menu: {
    isOpen: false,
    selectedTab: "PLAYLIST"
  },
  playlist: {
    previous: [],
    current: {
      _id: "5f14218d4ee369215e05624f",
      name: "Hey",
      artists: [
        {
          _id: "5f14228b4ee369215e05625a",
          name: "Benjamin Tissot"
        }
      ],
      albums: [
        {
          _id: "5f1423c54ee369215e05625e",
          name: "Sample Album A"
        }
      ]
    },
    next: [
      {
        _id: "5f1422054ee369215e056256",
        name: "Cute",
        artists: [
          {
            _id: "5f14228b4ee369215e05625a",
            name: "Benjamin Tissot"
          }
        ],
        albums: [
          {
            _id: "5f1423c54ee369215e05625e",
            name: "Sample Album A"
          }
        ]
      },
      {
        _id: "5f1422054ee369215e056257",
        name: "Going Higher",
        artists: [
          {
            _id: "5f14228b4ee369215e05625a",
            name: "Benjamin Tissot"
          }
        ],
        albums: [
          {
            _id: "5f1423c54ee369215e05625e",
            name: "Sample Album A"
          }
        ]
      },
      {
        _id: "5f1ad1ef17e08f691bb74048",
        name: "Creative Minds",
        artists: [
          {
            _id: "5f14228b4ee369215e05625a",
            name: "Benjamin Tissot"
          },
          {
            _id: "5f14228b4ee369215e05625b",
            name: "BenSound"
          }
        ],
        albums: [
          {
            _id: "5f1423c54ee369215e05625e",
            name: "Sample Album A"
          }
        ]
      }
    ]
  },
  playlists: {
    loading: false,
    data: {
      "5f1ffca0faf3a97662868ac0": {
        name: "Test Playlist 1",
        songs: [
          {
            _id: "5f1422054ee369215e056257",
            name: "Going Higher",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625e",
                name: "Sample Album A"
              }
            ]
          },
          {
            _id: "5f1ad1ef17e08f691bb74048",
            name: "Creative Minds",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              },
              {
                _id: "5f14228b4ee369215e05625b",
                name: "BenSound"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625e",
                name: "Sample Album A"
              }
            ]
          },
          {
            _id: "5f1422054ee369215e056258",
            name: "Ukulele",
            artists: [
              {
                _id: "5f14228b4ee369215e05625b",
                name: "BenSound"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625f",
                name: "Sample Album B"
              }
            ]
          },
          {
            _id: "5f1422054ee369215e056255",
            name: "A New Beginning",
            artists: [
              {
                _id: "5f14228b4ee369215e05625b",
                name: "BenSound"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625f",
                name: "Sample Album B"
              }
            ]
          }
        ]
      },
      "5f21389bc969de0689f10656": {
        name: "Test Playlist 2",
        songs: [
          {
            _id: "5f14218d4ee369215e05624f",
            name: "Hey",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625e",
                name: "Sample Album A"
              }
            ]
          },
          {
            _id: "5f1422054ee369215e056256",
            name: "Cute",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625e",
                name: "Sample Album A"
              }
            ]
          }
        ]
      },
      "5f213936c969de0689f10657": {
        name: "Test Playlist 3",
        songs: [
          {
            _id: "5f1422054ee369215e056258",
            name: "Ukulele",
            artists: [
              {
                _id: "5f14228b4ee369215e05625b",
                name: "BenSound"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625f",
                name: "Sample Album B"
              }
            ]
          }
        ]
      }
    },
    error: null
  },
  songs: {
    loading: false,
    data: [
      {
        _id: "5f1422054ee369215e056255",
        name: "A New Beginning",
        artists: [
          {
            _id: "5f14228b4ee369215e05625b",
            name: "BenSound"
          }
        ],
        albums: [
          {
            _id: "5f1423c54ee369215e05625f",
            name: "Sample Album B"
          }
        ]
      },
      {
        _id: "5f1ad1ef17e08f691bb74048",
        name: "Creative Minds",
        artists: [
          {
            _id: "5f14228b4ee369215e05625a",
            name: "Benjamin Tissot"
          },
          {
            _id: "5f14228b4ee369215e05625b",
            name: "BenSound"
          }
        ],
        albums: [
          {
            _id: "5f1423c54ee369215e05625e",
            name: "Sample Album A"
          }
        ]
      },
      {
        _id: "5f1422054ee369215e056256",
        name: "Cute",
        artists: [
          {
            _id: "5f14228b4ee369215e05625a",
            name: "Benjamin Tissot"
          }
        ],
        albums: [
          {
            _id: "5f1423c54ee369215e05625e",
            name: "Sample Album A"
          }
        ]
      },
      {
        _id: "5f1422054ee369215e056257",
        name: "Going Higher",
        artists: [
          {
            _id: "5f14228b4ee369215e05625a",
            name: "Benjamin Tissot"
          }
        ],
        albums: [
          {
            _id: "5f1423c54ee369215e05625e",
            name: "Sample Album A"
          }
        ]
      },
      {
        _id: "5f14218d4ee369215e05624f",
        name: "Hey",
        artists: [
          {
            _id: "5f14228b4ee369215e05625a",
            name: "Benjamin Tissot"
          }
        ],
        albums: [
          {
            _id: "5f1423c54ee369215e05625e",
            name: "Sample Album A"
          }
        ]
      },
      {
        _id: "5f1422054ee369215e056258",
        name: "Ukulele",
        artists: [
          {
            _id: "5f14228b4ee369215e05625b",
            name: "BenSound"
          }
        ],
        albums: [
          {
            _id: "5f1423c54ee369215e05625f",
            name: "Sample Album B"
          }
        ]
      }
    ],
    error: null
  }
};
