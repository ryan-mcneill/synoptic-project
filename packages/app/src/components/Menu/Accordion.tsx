import React, { FC, ReactElement, useState } from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faChevronDown, faChevronUp } from "@fortawesome/free-solid-svg-icons";
import { SongList } from "./";
import { Song } from "../../store/types";

interface AccordionProps {
  title: string;
  data: Song[];
}

const Accordion: FC<AccordionProps> = ({
  title,
  data
}): ReactElement => {
  const [isOpen, setIsOpen] = useState(false);

  return (
    <>
      <div
        style={{
          width: "100%",
          display: "flex",
          alignItems: "center",
          fontSize: "48px",
          padding: "1.5vh 0"
        }}
        onClick={() => setIsOpen(!isOpen)}
      >
        <div style={{ width: "calc(100% - 42px)" }}>{title}</div>
        <div style={{ width: "42px" }}>
          <FontAwesomeIcon icon={isOpen ? faChevronUp : faChevronDown} />
        </div>
      </div>

      <SongList isVisible={isOpen} data={data} />
    </>
  );
};

export default Accordion;
