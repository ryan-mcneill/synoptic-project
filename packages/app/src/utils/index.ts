import { artistsToString } from "./artistsToString";
import { filterData } from "./filterData";
import { filterSongs } from "./filterSongs";
import { playlistsToArray } from "./playlistsToArray";

export { artistsToString, filterData, filterSongs, playlistsToArray };
