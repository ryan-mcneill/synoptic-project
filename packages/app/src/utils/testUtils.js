import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { createStore } from "redux";
import { Provider } from "react-redux";
import reducers from "../store/reducers";

function render(
  ui,
  {
    initialState,
    store = createStore(reducers, initialState),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from "@testing-library/react";
export { render };
