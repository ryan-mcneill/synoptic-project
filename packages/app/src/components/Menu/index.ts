import Accordion from "./Accordion";
import InfoTile from "./InfoTile";
import PlaylistTab from "./PlaylistTab";
import SearchTab from "./SearchTab";
import SongList from "./SongList";
import SongMenu from "./SongMenu";
import SongsTab from "./SongsTab";
import TabButton from "./TabButton";
import Tabs from "./Tabs";

export {
  Accordion,
  InfoTile,
  PlaylistTab,
  SearchTab,
  SongList,
  SongMenu,
  SongsTab,
  TabButton,
  Tabs
};
