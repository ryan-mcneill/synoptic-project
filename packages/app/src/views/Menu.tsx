import React, { FC, ReactElement } from "react";
import { Action } from "redux";
import { connect, ConnectedProps } from "react-redux";
import { ThunkDispatch } from "redux-thunk";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faEllipsisH } from "@fortawesome/free-solid-svg-icons";
import { Tabs } from "../components/Menu";
import { setIsOpen } from "../store/actions";
import { State } from "../store/types";

const mapDispatchToProps = (dispatch: ThunkDispatch<State, void, Action>) => ({
  setIsOpen: (isOpen?: boolean) => dispatch(setIsOpen(isOpen))
});

const mapStateToProps = (state: State) => ({
  albumsLoading: state.albums.loading,
  artistsLoading: state.artists.loading,
  isOpen: state.menu.isOpen
});

const connector = connect(mapStateToProps, mapDispatchToProps);

type PropsFromRedux = ConnectedProps<typeof connector>;

const Menu: FC<PropsFromRedux> = ({
  albumsLoading,
  artistsLoading,
  isOpen,
  setIsOpen
}): ReactElement =>
  !albumsLoading && !artistsLoading ? (
    <div
      style={{
        position: "absolute",
        bottom: isOpen ? 0 : "-90vh",
        transition: "all 0.5s ease"
      }}
    >
      <div
        style={{
          textAlign: "center",
          fontSize: "72px",
          position: "relative",
          left: "50%",
          bottom: "-1px",
          transform: "translateX(-50%)",
          height: "72px",
          width: "128px",
          backgroundColor: "rgba(255, 255, 255, .5)",
          borderRadius: "24px 24px 0 0",
          backdropFilter: "blur(20px)"
        }}
        onClick={() => setIsOpen()}
      >
        <FontAwesomeIcon icon={faEllipsisH} />
      </div>
      <Tabs />
    </div>
  ) : (
    <></>
  );

export default connector(Menu);
