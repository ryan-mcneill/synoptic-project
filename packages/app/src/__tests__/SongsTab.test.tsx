import React from "react";
import { SongsTab } from "../components/Menu";
import { render, fireEvent } from "../utils/testUtils";
import { mockState } from "../__mocks__/state";

describe("The SongsTab component", () => {
  const props = {
    artists: mockState.artists.data,
    albums: mockState.albums.data,
    playlists: mockState.playlists.data,
    songs: mockState.songs.data
  };

  it("renders correctly on the SONGS tab", () => {
    const { container } = render(<SongsTab {...props} />, {
      initialState: mockState
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly on the ARTISTS tab", () => {
    const { container, getByText } = render(<SongsTab {...props} />, {
      initialState: mockState
    });

    const tabButton = getByText("ARTISTS");
    fireEvent.click(tabButton);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly on the ALBUMS tab", () => {
    const { container, getByText } = render(<SongsTab {...props} />, {
      initialState: mockState
    });

    const tabButton = getByText("ALBUMS");
    fireEvent.click(tabButton);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly on the PLAYLISTS tab", () => {
    const { container, getByText } = render(<SongsTab {...props} />, {
      initialState: mockState
    });

    const tabButton = getByText("PLAYLISTS");
    fireEvent.click(tabButton);

    expect(container.firstChild).toMatchSnapshot();
  });
});
