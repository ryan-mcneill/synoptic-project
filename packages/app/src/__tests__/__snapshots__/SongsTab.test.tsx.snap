// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The SongsTab component renders correctly on the ALBUMS tab 1`] = `
<div
  style="height: inherit;"
>
  <div
    style="display: flex; align-items: center; justify-content: space-around; font-size: 36px; width: 100%; text-align: center; margin-bottom: 20px;"
  >
    <div
      title="SONGS"
    >
      <span>
        SONGS
      </span>
      <div
        style="width: 0px; height: 8px; background-color: black; position: relative; left: 0px; margin-top: 8px; transition: all 0.5s ease;"
      />
    </div>
    <div
      title="ARTISTS"
    >
      <span>
        ARTISTS
      </span>
      <div
        style="width: 0px; height: 8px; background-color: black; position: relative; left: 0px; margin-top: 8px; transition: all 0.5s ease;"
      />
    </div>
    <div
      title="ALBUMS"
    >
      <span>
        ALBUMS
      </span>
      <div
        style="width: 100%; height: 8px; background-color: black; position: relative; left: 0px; margin-top: 8px; transition: all 0.5s ease;"
      />
    </div>
    <div
      title="PLAYLISTS"
    >
      <span>
        PLAYLISTS
      </span>
      <div
        style="width: 0px; height: 8px; background-color: black; position: relative; left: 0px; margin-top: 8px; transition: all 0.5s ease;"
      />
    </div>
  </div>
  <div
    style="overflow-y: scroll; height: 100%;"
  >
    <div
      style="width: 100%; display: flex; align-items: center; font-size: 48px; padding: 1.5vh 0px;"
    >
      <div>
        Sample Album A
      </div>
      <div
        style="width: 42px;"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-chevron-down fa-w-14 "
          data-icon="chevron-down"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div
      style="width: 100%; display: flex; align-items: center; font-size: 48px; padding: 1.5vh 0px;"
    >
      <div>
        Sample Album B
      </div>
      <div
        style="width: 42px;"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-chevron-down fa-w-14 "
          data-icon="chevron-down"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`The SongsTab component renders correctly on the ARTISTS tab 1`] = `
<div
  style="height: inherit;"
>
  <div
    style="display: flex; align-items: center; justify-content: space-around; font-size: 36px; width: 100%; text-align: center; margin-bottom: 20px;"
  >
    <div
      title="SONGS"
    >
      <span>
        SONGS
      </span>
      <div
        style="width: 0px; height: 8px; background-color: black; position: relative; left: 0px; margin-top: 8px; transition: all 0.5s ease;"
      />
    </div>
    <div
      title="ARTISTS"
    >
      <span>
        ARTISTS
      </span>
      <div
        style="width: 100%; height: 8px; background-color: black; position: relative; left: 0px; margin-top: 8px; transition: all 0.5s ease;"
      />
    </div>
    <div
      title="ALBUMS"
    >
      <span>
        ALBUMS
      </span>
      <div
        style="width: 0px; height: 8px; background-color: black; position: relative; left: 0px; margin-top: 8px; transition: all 0.5s ease;"
      />
    </div>
    <div
      title="PLAYLISTS"
    >
      <span>
        PLAYLISTS
      </span>
      <div
        style="width: 0px; height: 8px; background-color: black; position: relative; left: 0px; margin-top: 8px; transition: all 0.5s ease;"
      />
    </div>
  </div>
  <div
    style="overflow-y: scroll; height: 100%;"
  >
    <div
      style="width: 100%; display: flex; align-items: center; font-size: 48px; padding: 1.5vh 0px;"
    >
      <div>
        Benjamin Tissot
      </div>
      <div
        style="width: 42px;"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-chevron-down fa-w-14 "
          data-icon="chevron-down"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div
      style="width: 100%; display: flex; align-items: center; font-size: 48px; padding: 1.5vh 0px;"
    >
      <div>
        BenSound
      </div>
      <div
        style="width: 42px;"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-chevron-down fa-w-14 "
          data-icon="chevron-down"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`The SongsTab component renders correctly on the PLAYLISTS tab 1`] = `
<div
  style="height: inherit;"
>
  <div
    style="display: flex; align-items: center; justify-content: space-around; font-size: 36px; width: 100%; text-align: center; margin-bottom: 20px;"
  >
    <div
      title="SONGS"
    >
      <span>
        SONGS
      </span>
      <div
        style="width: 0px; height: 8px; background-color: black; position: relative; left: 0px; margin-top: 8px; transition: all 0.5s ease;"
      />
    </div>
    <div
      title="ARTISTS"
    >
      <span>
        ARTISTS
      </span>
      <div
        style="width: 0px; height: 8px; background-color: black; position: relative; left: 0px; margin-top: 8px; transition: all 0.5s ease;"
      />
    </div>
    <div
      title="ALBUMS"
    >
      <span>
        ALBUMS
      </span>
      <div
        style="width: 0px; height: 8px; background-color: black; position: relative; left: 0px; margin-top: 8px; transition: all 0.5s ease;"
      />
    </div>
    <div
      title="PLAYLISTS"
    >
      <span>
        PLAYLISTS
      </span>
      <div
        style="width: 100%; height: 8px; background-color: black; position: relative; left: 0px; margin-top: 8px; transition: all 0.5s ease;"
      />
    </div>
  </div>
  <div
    style="overflow-y: scroll; height: 100%;"
  >
    <div
      style="width: 100%; display: flex; align-items: center; font-size: 48px; padding: 1.5vh 0px;"
    >
      <div>
        Test Playlist 1
      </div>
      <div
        style="width: 42px;"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-chevron-down fa-w-14 "
          data-icon="chevron-down"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div
      style="width: 100%; display: flex; align-items: center; font-size: 48px; padding: 1.5vh 0px;"
    >
      <div>
        Test Playlist 2
      </div>
      <div
        style="width: 42px;"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-chevron-down fa-w-14 "
          data-icon="chevron-down"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div
      style="width: 100%; display: flex; align-items: center; font-size: 48px; padding: 1.5vh 0px;"
    >
      <div>
        Test Playlist 3
      </div>
      <div
        style="width: 42px;"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-chevron-down fa-w-14 "
          data-icon="chevron-down"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`The SongsTab component renders correctly on the SONGS tab 1`] = `
<div
  style="height: inherit;"
>
  <div
    style="display: flex; align-items: center; justify-content: space-around; font-size: 36px; width: 100%; text-align: center; margin-bottom: 20px;"
  >
    <div
      title="SONGS"
    >
      <span>
        SONGS
      </span>
      <div
        style="width: 100%; height: 8px; background-color: black; position: relative; left: 0px; margin-top: 8px; transition: all 0.5s ease;"
      />
    </div>
    <div
      title="ARTISTS"
    >
      <span>
        ARTISTS
      </span>
      <div
        style="width: 0px; height: 8px; background-color: black; position: relative; left: 0px; margin-top: 8px; transition: all 0.5s ease;"
      />
    </div>
    <div
      title="ALBUMS"
    >
      <span>
        ALBUMS
      </span>
      <div
        style="width: 0px; height: 8px; background-color: black; position: relative; left: 0px; margin-top: 8px; transition: all 0.5s ease;"
      />
    </div>
    <div
      title="PLAYLISTS"
    >
      <span>
        PLAYLISTS
      </span>
      <div
        style="width: 0px; height: 8px; background-color: black; position: relative; left: 0px; margin-top: 8px; transition: all 0.5s ease;"
      />
    </div>
  </div>
  <div
    style="overflow-y: scroll; height: 100%;"
  >
    <div
      style="margin-top: 80px;"
    >
      <div
        style="display: block;"
      >
        <div
          style="font-size: 48px; margin: 40px 0px;"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-play fa-w-14 "
            data-icon="play"
            data-prefix="fas"
            focusable="false"
            role="img"
            style="margin-left: 8px; margin-right: 40px;"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M424.4 214.7L72.4 6.6C43.8-10.3 0 6.1 0 47.9V464c0 37.5 40.7 60.1 72.4 41.3l352-208c31.4-18.5 31.5-64.1 0-82.6z"
              fill="currentColor"
            />
          </svg>
          Play all
        </div>
        <div>
          <div
            style="width: 100%; display: flex; align-items: center; font-size: 48px; padding: 2vh 0px;"
          >
            <div
              style="width: 57px; height: auto; padding-right: 32px;"
            />
            <div>
              <div
                style="font-size: 48px;"
              >
                A New Beginning
              </div>
              <div
                style="font-size: 36px;"
              >
                BenSound
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            style="width: 100%; display: flex; align-items: center; font-size: 48px; padding: 2vh 0px;"
          >
            <div
              style="width: 57px; height: auto; padding-right: 32px;"
            />
            <div>
              <div
                style="font-size: 48px;"
              >
                Creative Minds
              </div>
              <div
                style="font-size: 36px;"
              >
                Benjamin Tissot, BenSound
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            style="width: 100%; display: flex; align-items: center; font-size: 48px; padding: 2vh 0px;"
          >
            <div
              style="width: 57px; height: auto; padding-right: 32px;"
            />
            <div>
              <div
                style="font-size: 48px;"
              >
                Cute
              </div>
              <div
                style="font-size: 36px;"
              >
                Benjamin Tissot
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            style="width: 100%; display: flex; align-items: center; font-size: 48px; padding: 2vh 0px;"
          >
            <div
              style="width: 57px; height: auto; padding-right: 32px;"
            />
            <div>
              <div
                style="font-size: 48px;"
              >
                Going Higher
              </div>
              <div
                style="font-size: 36px;"
              >
                Benjamin Tissot
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            style="width: 100%; display: flex; align-items: center; font-size: 48px; padding: 2vh 0px;"
          >
            <div
              style="width: 57px; height: auto; padding-right: 32px;"
            />
            <div>
              <div
                style="font-size: 48px;"
              >
                Hey
              </div>
              <div
                style="font-size: 36px;"
              >
                Benjamin Tissot
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            style="width: 100%; display: flex; align-items: center; font-size: 48px; padding: 2vh 0px;"
          >
            <div
              style="width: 57px; height: auto; padding-right: 32px;"
            />
            <div>
              <div
                style="font-size: 48px;"
              >
                Ukulele
              </div>
              <div
                style="font-size: 36px;"
              >
                BenSound
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
